packages/split-layout/src/vaadin-split-layout-mixin.js

```diff
@@ -63,17 +63,34 @@ export const SplitLayoutMixin = (superClass) =>
 
     /** @private */
     _processChildren() {
-      [...this.children].forEach((child, i) => {
-        if (i === 0) {
-          this._primaryChild = child;
-          child.setAttribute('slot', 'primary');
-        } else if (i === 1) {
-          this._secondaryChild = child;
-          child.setAttribute('slot', 'secondary');
-        } else {
+      this.querySelectorAll('[slot]').forEach((child) => {
```

Comment:
Using `querySelectorAll` here is problematic with nested split layouts, e.g. the following doesn't work:

```html
<vaadin-split-layout style="width: 400px; height: 200px">
  <vaadin-split-layout style="width: 200px; height: 200px" orientation="vertical">
    <div slot="primary">foo</div>
    <div slot="secondary">bar</div>
  </vaadin-split-layout>
  <div>baz</div>
</vaadin-split-layout>
```